File: CPP/Problem_024.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

static long long factorial(int n) {
    long long f = 1;
    for (int i = 2; i <= n; ++i) f *= i;
    return f;
}

int main() {
    vector<char> digits = {'0','1','2','3','4','5','6','7','8','9'};
    long long k = 1'000'000 - 1; // 0-based

    string result;
    result.reserve(10);

    for (int n = (int)digits.size(); n >= 1; --n) {
        long long f = factorial(n - 1);
        long long idx = k / f;
        k %= f;

        result.push_back(digits[(int)idx]);
        digits.erase(digits.begin() + (int)idx);
    }

    cout << result << "\n"; // 2783915460
    return 0;
}

File: Python/Problem_024.py
import math

def solve() -> str:
    digits = list("0123456789")
    k = 1_000_000 - 1  # 0-based
    out = []

    for n in range(len(digits), 0, -1):
        f = math.factorial(n - 1)
        idx, k = divmod(k, f)
        out.append(digits.pop(idx))

    return "".join(out)

if __name__ == "__main__":
    print(solve())  # 2783915460
